src/circuit/v2/util.ts

```diff
@@ -16,85 +16,102 @@ const doRelay = (src: DuplexStream, dst: DuplexStream) => {
   })
 }
 
-export function createLimitedRelay (source: Abortable<DuplexStream>, destination: Abortable<DuplexStream>, limit?: Limit) {
+export function createLimitedRelay (source: Stream, destination: Stream, limit?: Limit) {
   // trivial case
   if (limit == null) {
-    doRelay(source.value, destination.value)
+    doRelay(source, destination)
     return
   }
-  const dataLimit = limit.data ?? BigInt(0)
+
+  const dataLimit = limit.data ?? 0n
   const durationLimit = limit.duration ?? 0
   const src = durationLimitDuplex(dataLimitDuplex(source, dataLimit), durationLimit)
   const dst = durationLimitDuplex(dataLimitDuplex(destination, dataLimit), durationLimit)
-  doRelay(src.value, dst.value)
+
+  doRelay(src, dst)
 }
 
-const dataLimitSource = (source: Source<Uint8ArrayList>, abort: (err: Error) => void, limit: bigint): Source<Uint8ArrayList> => {
-  if (limit === BigInt(0)) {
-    return source
+const dataLimitSource = (stream: Stream, limit: bigint): Stream => {
```

Comment:
Is this safe? While the implementation of `it-pb-stream` returns the underlying stream, the return type only guarantees returning a `Duplex`. The stream functions could be erased from the returned value in a future release without changing the API, which would cause this to break.

Reply:
`3.0.1` changes the `unwrap` return type to be the type of the wrapped stream, so it should be safe.